Source/JavaScriptCore/heap/FreeList.h

```diff
@@ -94,12 +105,12 @@ class FreeList {
     unsigned cellSize() const { return m_cellSize; }
     
 private:
-    FreeCell* head() const { return FreeCell::descramble(m_scrambledHead, m_secret); }
+    FreeCell* head() const { return m_head; }
     
-    uintptr_t m_scrambledHead { 0 };
-    uintptr_t m_secret { 0 };
+    char* m_payloadStart { nullptr };
     char* m_payloadEnd { nullptr };
-    unsigned m_remaining { 0 };
+    FreeCell* m_head { reinterpret_cast_ptr<FreeCell*>(1) };
```

Comment:
Let's name this `m_nextInterval` instead of `m_head`.  `m_head` implies this is where we start, which is ambiguous if one is thinking of where we start allocating.  `m_nextInterval` (paired with `m_intervalStart` and `m_intervalEnd`) makes it clear what this is really for.

Reply:
Resolved in latest push.